client/src/components/ProtectedRoute/ProtectedRoute.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { useEffect } from 'react';
+import { useAuth } from '../../context/useAuthContext';
+import { useSocket } from '../../context/useSocketContext';
+import CircularProgress from '@mui/material/CircularProgress';
+import { Route, Redirect, RouteProps, RouteComponentProps } from 'react-router-dom';
+
+const ProtectedRoute: React.FC<RouteProps> = ({ component, render, ...routeProps }) => {
+  const { loggedInUser } = useAuth();
+  const { initSocket } = useSocket();
+
+  useEffect(() => {
+    initSocket();
+  }, [initSocket]);
+
+  if (loggedInUser === undefined) {
+    // show progress circule until we have some value for user
+    return <CircularProgress />;
+  }
+  if (!loggedInUser) {
+    // redirect if user is not authenticated
+    return (
+      <Redirect
+        to={{
+          pathname: '/login',
+          state: {
+            from: routeProps.location,
+          },
+        }}
+      />
+    );
+  }
+
+  return (
+    <Route
```

Comment:
```suggestion
    return <Route {... routeProps} render={(props) => (loggedInUser ? <Component {...props} /> : <Redirect to="/login" />)} />;
```

We can simplify this component further!